pkg/storage/segments/store.go

```diff
@@ -76,30 +155,59 @@ func (s *segmentStore) Put(ctx context.Context, path paths.Path, data io.Reader,
 	metadata []byte, expiration time.Time) (meta Meta, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	// uses overlay client to request a list of nodes
-	nodes, err := s.oc.Choose(ctx, s.rs.TotalCount(), 0)
+	var p *ppb.Pointer
+
+	peekReader := NewPeekThresholdReader(data)
+	remoteSized, err := peekReader.isInline(s.thresholdSize)
 	if err != nil {
-		return Meta{}, Error.Wrap(err)
+		return Meta{}, err
+	}
+	if !remoteSized {
+		p = &ppb.Pointer{
+			Type:     ppb.Pointer_INLINE,
+			Metadata: metadata,
+		}
+	} else {
+		// uses overlay client to request a list of nodes
+		nodes, err := s.oc.Choose(ctx, s.rs.TotalCount(), 0)
+		if err != nil {
+			return Meta{}, Error.Wrap(err)
+		}
+		pieceID := client.NewPieceID()
+
+		// puts file to ecclient
+		err = s.ec.Put(ctx, nodes, s.rs, pieceID, data, expiration)
```

Comment:
We should pass `peekReader` instead of `data`. Otherwise ECClient won't be able to read the first "threshold" bytes.